src/workloads/execution/lookup/LookupNLJ.yml

```diff
@@ -0,0 +1,345 @@
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kScalarArr8Cmd
+- OperationMetricsName: Foreign1k_Array8_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign1kArr8ScalarCmd
+- OperationMetricsName: Foreign10k_Scalar_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarScalarCmd
+- OperationMetricsName: Foreign10k_Scalar_Object2
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarObj2Cmd
+- OperationMetricsName: Foreign10k_Scalar_Object5
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarObj5Cmd
+- OperationMetricsName: Foreign10k_Object5_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kObj5ScalarCmd
+- OperationMetricsName: Foreign10k_Scalar_Array1
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarArr1Cmd
+- OperationMetricsName: Foreign10k_Scalar_Array8
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kScalarArr8Cmd
+- OperationMetricsName: Foreign10k_Array8_Scalar
+  OperationName: RunCommand
+  OperationCommand: *NLJLookupForeign10kArr8ScalarCmd
+
+# Defines how many times each test operation is executed.
+TestRepeatCount: &TestRepeatCount 10
+
+Actors:
+# The 'Setup' actor drops the database and runs in the 0th phase.
+- Name: Setup
+  Type: RunCommand
+  Database: *db
+  Thread: 1
+  Phases:
+  - Repeat: 1
+    Threads: 1
+    Database: *db
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand: {dropDatabase: 1}
+  - Phase: 1..3
+    Nop: true
+
+# The 'LoadForeign1K' / 'LoadForeign10K' / 'LoadLocalDoc100k' actors run in the 1st phase.
+- Name: LoadForeign1K
+  Type: CrudActor
+  Database: *db
+  Threads: 1
+  Phases:
+  - &Nop {Nop: true}
+  - Repeat: 1000
+    Threads: 1
+    Collection: *Foreign1K
+    Operations:
+    - OperationName: insertOne
+      OperationCommand:
+        Document: *LoadDoc
+  - *Nop
+  - *Nop
+
+- Name: LoadForeign10K
+  Type: CrudActor
+  Database: *db
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 10000
+    Threads: 1
+    Collection: *Foreign10K
+    Operations:
+    - OperationName: insertOne
+      OperationCommand:
+        Document: *LoadDoc
+  - *Nop
+  - *Nop
+
+- Name: LoadLocalDoc100k
+  Type: CrudActor
+  Database: *db
+  Threads: 10
+  Phases:
+  - *Nop
+  - Repeat: 10000
+    Threads: 1
+    Collection: *LocalDoc100k
+    Operations:
+    - OperationName: insertOne
+      OperationCommand:
+        Document: *LoadDoc
+  - *Nop
+  - *Nop
+
+# The 'Quiesce' actor calms down to avoid any trailing impacts from the previous phase and runs
+# in the 2nd phase.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+# The 'NLJLookup' actor measures the NLJ lookup performances and runs in the 3rd phase.
+- Name: NLJLookup
+  Type: RunCommand
+  Database: *db
+  Phases:
+  - Phase: 0..2
```

Comment:
[question] Am I correct that this is saying 'noop during phases 0 though 2', similar to how the setup above is a no op from phases 1 to 3?

Reply:
yes.